### ISSUE TITLE
disconnect doesn't work if triggered by emit

### ISSUE THREAD

**Issue by iamsergio** [voidari/flutter_signals_slots#4]
Here's a minimal example, contrived, but just to illustrate the point:

```dart
import 'package:signals_slots/signals_slots.dart';

class Foo {
  late final Connection conn;
  final sig = Signal0();
  Foo() {
    conn = sig.connect(() {
      conn.disconnect();
    });
  }
}

void main() {
  final foo = Foo();
  foo.sig.emit();
}

```

Results in:

```bash
❯ dart main.dart
Unhandled exception:
Concurrent modification during iteration: Instance(length:0) of '_GrowableList'.
#0      ListIterator.moveNext (dart:_internal/iterable.dart:348:7)
#1      Signal0.emit (package:signals_slots/src/signal.dart:130:58)
#2      main (file:///Users/serj/lixo/sigtest/main.dart:15:11)
#3      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:297:19)
#4      _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)
```

Could this be supported ?

**Comment by iamsergio** [comment]
inside emit(), probably iterate on a copy of `_connectionMap[group]!`  instead

**Comment by graybrandonh** [comment]
@iamsergio thank you for the feedback! I'll take a look at this issue later today and resolve it as soon as I can.

**Comment by graybrandonh** [comment]
@iamsergio I've released 1.1.1 with changes to fix your issue. Please close out this issue after verifying it fixes your problem.

**Comment by iamsergio** [comment]
Problem fixed, thanks !
